pandas/tests/indexes/test_category.py

```diff
@@ -1123,3 +1123,25 @@ def test_engine_type(self, dtype, engine_type):
             ci.values._codes = ci.values._codes.astype("int64")
         assert np.issubdtype(ci.codes.dtype, dtype)
         assert isinstance(ci._engine, engine_type)
+
+    def test_dt_tz_localize(self, tz_aware_fixture):
+        # GH 27952
+        tz = tz_aware_fixture
+        datetimes = pd.Series(
+            ["2019-01-01", "2019-01-01", "2019-01-02"], dtype="datetime64[ns]"
+        )
+        categorical = datetimes.astype("category")
+        result = categorical.dt.tz_localize(tz)
+        expected = datetimes.dt.tz_localize(tz)
+        tm.assert_series_equal(result, expected)
```

Comment:
can you do the same with tz_convert (starting from a tz-aware dtype)?

Reply:
should the result here be datetime64tz or categorical[datetime64tz]?  My initial intuition is that it would stay categorical

Reply:
@jbrockmendel thanks for your feedback! I've made the changes you suggested.

Regarding the result dtype - the example from #27952 returns
```
dtype: datetime64[ns]
```
and so this isn't affected by my changes. Perhaps it should be a separate issue?

Reply:
Also, the docs say
> Note The returned Series (or DataFrame) is of the same type as if you used the .str.<method> / .dt.<method> on a Series of that type (and not of type category!).

so this is consistent with that - but I'm happy to work on changing this if that's what you prefer